docs/contributing/coding-standards/component-options.md

```diff
@@ -4,204 +4,42 @@ Configuration options allow developers to pass additional information into a com
 
-If it doesn't, you can return a `ConfigError`.
+- [defaults defined as a `defaults` static property](https://frontend.design-system.service.gov.uk/building-configurable-components/#setting-a-default-configuration-for-your-component) on the component’s class
+- [configuration passed as second argument](https://frontend.design-system.service.gov.uk/building-configurable-components/#receiving-configuration-during-initialisation) to the component’s constructor or [`createAll`](https://frontend.design-system.service.gov.uk/building-your-own-javascript-components/#using-createall-with-your-components)
+- [data attributes on the root element](https://frontend.design-system.service.gov.uk/building-configurable-components/#receiving-configuration-from-data-attributes) of the component
 
-```mjs
-import { mergeConfigs, validateConfig, normaliseDataset } from '../../common/configuration.mjs'
-import { ConfigError } from '../../errors/index.mjs'
-import { GOVUKFrontendComponent } from '../../govuk-frontend-component.mjs'
+These are the same sources as those used by GOV.UK Frontend components. Configuration options from these sources are merged into a single configuration object by the `ConfigurableComponent` class. Note that:
 
-export class Accordion extends GOVUKFrontendComponent {
-  constructor($root, config = {}) {
-    super($root)
+- configuration options passed to the constructor will override any defaults
+- data attribute values have highest precedence and will override any configuration set elsewhere
 
-    this.config = mergeConfigs(
-      Accordion.defaults,
-      config,
-      normaliseDataset(Accordion, this.$root.dataset)
-    )
+Your component will then be able to:
 
-    // Check that the configuration provided is valid
-    const errors = validateConfig(Accordion.schema, this.config)
-    if (errors[0]) {
-      throw new ConfigError(formatErrorMessage(CharacterCount, errors[0]))
-    }
-  }
-}
-```
+[access the merged configuration](https://frontend.design-system.service.gov.uk/building-configurable-components/#accessing-the-component-s-configuration) using `this.config`
+use [the`Component` class](https://frontend.design-system.service.gov.uk/building-your-own-javascript-components/#using-the-component-class) features
 
 ## Adding a Nunjucks parameter
 
-Most, but not all, components support adding arbitrary attributes and values through the `attributes` parameter. This method is also more verbose compared to having a dedicated Nunjucks parameter.
-
-Using the previous bit of HTML, update it to make the `data-*` attribute changeable.
+If you're using Nunjucks macros, you can use a dedicated Nunjucks parameter to set data attributes on the root element.
 
 ```nunjucks
 <div
-  class="govuk-accordion"
-  data-module="govuk-accordion"
+  class="my-component"
+  data-module="my-component"
```

Comment:
**question** As these are internal docs, could we stick with the 'govuk' prefix? Maybe it's a matter of changing the introduction for the example, leading with something like:

'For example, this is how the Accordion does it for its `rememberExpanded` option:'